app/controllers/staff/transactions_controller.rb

```diff
@@ -30,23 +34,26 @@ def create
 
   def edit
     @transaction = policy_scope(Transaction).find(id)
-    @fund = Fund.find(fund_id)
     authorize @transaction
+
+    @fund = Fund.find(fund_id)
   end
 
   def update
     @transaction = policy_scope(Transaction).find(id)
     authorize @transaction
 
+    @fund = Fund.find(fund_id)
```

Comment:
The edit form needs this as an instance variable which is rendered when the edit form is invalid.